### PR TITLE
Blockaid mobile performance improvements

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

The PR implements performance improvements for blockaid mobile.
Ref PR: https://github.com/MMBAPPOM/metamask-mobile/pull/1
Performance improvement is achieved by:
1. using native cryptography
2. using `react-native-mmkv` for file storage

## **Related issues**

Fixes: MetaMask/MetaMask-planning#1937

## **Manual testing steps**

1. Open the app and enable blockaid
2. Switch between various networks supported by blockaid
3. Submit malicious transactions to ensure they are correctly validated

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [X] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
I am looking at the broken build.

**Comment by jpuri** [comment]
@SocketSecurity ignore react-native-quick-crypto@0.6.1
@SocketSecurity ignore @metamask/ppom-validator@0.22.0

File: yarn.lock [hunk 1]
```diff
@@ -10701,7 +10709,7 @@ browserify-rsa@^4.0.0, browserify-rsa@^4.1.0:
     bn.js "^5.0.0"
     randombytes "^2.0.1"
 
-browserify-sign@4.2.2, browserify-sign@^4.0.4:
+browserify-sign@4.2.2, browserify-sign@^4.0.0, browserify-sign@^4.0.4:
```
Comment (naugtur) [inline yarn.lock:10712]: versions diverged unnecesarily. can be deduped

Comment (jpuri) [reply yarn.lock:10712]: Running dedupe command seems to have no effect on this.

**Comment by jpuri** [comment]
@SocketSecurity ignore @metamask/ppom-validator/0.22.0
@SocketSecurity ignore react-native-quick-crypto/0.6.1

**Comment by jpuri** [comment]
@SocketSecurity ignore @metamask/ppom-validator@0.22.0
@SocketSecurity ignore react-native-quick-crypto@0.6.1

**Review by segun — APPROVED** [review]

File: jestSetup.js
Comment (leotm) [inline jestSetup.js:1]: can we move these to https://github.com/MetaMask/metamask-mobile/blob/4e2a83fa0d9f15351e395c7cf45b25096443d0b0/jest.config.js#L6?

File: jest.config.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ const config = {
     '\\.svg': '<rootDir>/app/__mocks__/svgMock.js',
     '\\.png': '<rootDir>/app/__mocks__/pngMock.js',
   },
+  setupFiles: ['./jestSetup.js'],
```
Comment (leotm) [inline jest.config.js:33]: think we can move our 2 new mocks to

https://github.com/MetaMask/metamask-mobile/blob/4e2a83fa0d9f15351e395c7cf45b25096443d0b0/jest.config.js#L6

unless they need to be run before `setupFilesAfterEnv`

Comment (jpuri) [reply jest.config.js:33]: Let me do this @leotm

Comment (jpuri) [reply jest.config.js:33]: I foxed it here @leotm https://github.com/MetaMask/metamask-mobile/pull/8351

Comment (leotm) [reply jest.config.js:33]: thx ^^

File: jestSetup.js
Comment (leotm) [reply jestSetup.js:1]: resolved in: https://github.com/MetaMask/metamask-mobile/pull/8343#discussion_r1459419890

**Pull request merged**